src/Capability/Registry/PromptReference.php

```diff
@@ -55,228 +47,6 @@ public function __construct(
-    {
-        if (!isset($content['text']) || !\is_string($content['text'])) {
-            throw new RuntimeException(\sprintf('Invalid "text" content%s: Missing or invalid "text" string.', $indexStr));
-        }
-
-        return new TextContent($content['text']);
-    }
-
-    /**
-     * @param array<string, mixed> $content
-     */
-    private function formatImageContent(array $content, string $indexStr): ImageContent
-    {
-        if (!isset($content['data']) || !\is_string($content['data'])) {
-            throw new RuntimeException("Invalid 'image' content{$indexStr}: Missing or invalid 'data' string (base64).");
-        }
-        if (!isset($content['mimeType']) || !\is_string($content['mimeType'])) {
-            throw new RuntimeException("Invalid 'image' content{$indexStr}: Missing or invalid 'mimeType' string.");
-        }
-
-        return new ImageContent($content['data'], $content['mimeType']);
-    }
-
-    /**
-     * @param array<string, mixed> $content
-     */
-    private function formatAudioContent(array $content, string $indexStr): AudioContent
-    {
-        if (!isset($content['data']) || !\is_string($content['data'])) {
-            throw new RuntimeException("Invalid 'audio' content{$indexStr}: Missing or invalid 'data' string (base64).");
-        }
-        if (!isset($content['mimeType']) || !\is_string($content['mimeType'])) {
-            throw new RuntimeException("Invalid 'audio' content{$indexStr}: Missing or invalid 'mimeType' string.");
-        }
-
-        return new AudioContent($content['data'], $content['mimeType']);
-    }
-
-    /**
-     * @param array<string, mixed> $content
-     */
-    private function formatResourceContent(array $content, string $indexStr): EmbeddedResource
-    {
-        if (!isset($content['resource']) || !\is_array($content['resource'])) {
-            throw new RuntimeException("Invalid 'resource' content{$indexStr}: Missing or invalid 'resource' object.");
-        }
-
-        $resource = $content['resource'];
-        if (!isset($resource['uri']) || !\is_string($resource['uri'])) {
-            throw new RuntimeException("Invalid resource{$indexStr}: Missing or invalid 'uri'.");
-        }
-
-        if (isset($resource['text']) && \is_string($resource['text'])) {
-            $resourceObj = new TextResourceContents($resource['uri'], $resource['mimeType'] ?? 'text/plain', $resource['text']);
-        } elseif (isset($resource['blob']) && \is_string($resource['blob'])) {
-            $resourceObj = new BlobResourceContents(
-                $resource['uri'],
-                $resource['mimeType'] ?? 'application/octet-stream',
-                $resource['blob']
-            );
-        } else {
-            throw new RuntimeException("Invalid resource{$indexStr}: Must contain 'text' or 'blob'.");
-        }
-
-        return new EmbeddedResource($resourceObj);
+        return (new PromptResultFormatter())->format($promptGenerationResult);
```

Comment:
I like that this was moved to a shared class, but I dont like that we create new objects here. 

But let's leave it for now